### PR TITLE
Feature/milestone5

### PR DESCRIPTION
**Pull request by trox115**
**Ready for Review**
- Implemented the clickhandlers function
- The totals and nextvalues showing in display
- Division by 0 showing too
- Buttons Listening to click

### **Edit:** If you don't want to look in the Readme here is the [live version](https://antonioreact-calculator.herokuapp.com/)

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -2,13 +2,42 @@ import React from 'react';
...
+    let value = null;
+    if (operation === null) {
+      value = total || '0';
+    } else {
+      value = next || total;
+    }
+    return (
+      <div id="main-div">
+        <Display value={value.toString()} />
+        <ButtonPanel onClick={btname => this.handle(btname)} />
```
Comment (Oghenebrume50) [inline src/components/App.js]: there should be a prop that handles the clickhandler, you are calling onclick directly that is not the desired behaviour screen

File: src/components/App.js [hunk 2]
```diff
@@ -2,13 +2,42 @@ import React from 'react';
...
+    const { next, operation, total } = this.state;
+    let value = null;
+    if (operation === null) {
+      value = total || '0';
+    } else {
+      value = next || total;
+    }
+    return (
+      <div id="main-div">
+        <Display value={value.toString()} />
```
Comment (Oghenebrume50) [inline src/components/App.js:37]: pass the properties(total and next)  in the state to Display screen

File: src/components/ButtonPanel.js [hunk 3]
```diff
@@ -1,38 +1,125 @@
...
-      <Button btname="." type="Button" color="gray" />
-      <Button btname="=" type="Button" />
-    </div>
-  </div>
-);
+class ButtonPanel extends React.Component {
+  handle(btname) {
+    const prps = this.props;
+    const btName = btname;
+    /* eslint react/prop-types: 0 */
```
Comment (Oghenebrume50) [inline src/components/ButtonPanel.js]: having inline instructions for ESlint is not encouraged, please try to eliminate the issue

File: src/components/ButtonPanel.js [hunk 4]
```diff
@@ -1,38 +1,125 @@
...
+
+  render() {
+    return (
+      <div className="Buttons">
+        <div className="group1">
+          <Button
+            btname="AC"
+            type="Button"
+            color="gray"
+            onClick={() => this.handle('AC')}
```
Comment (Oghenebrume50) [inline src/components/ButtonPanel.js]: we want to call onClick in the actual Button component not pass it as a prop for better semantics screen

File: src/components/Button.js [hunk 5]
```diff
@@ -3,23 +3,24 @@ import PropTypes from 'prop-types';
 
 const Button = (props) => {
```
Comment (Oghenebrume50) [inline src/components/Button.js]: there should be a function that calls the clickhandler prop screen

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
### STATUS: CHANGES REQUIRED

Hi @trox115 good job on this one, please see the suggested changes below, they are basically semantics but, they are paramount to the react logic. 
**please make changes and request for another review**

**RAPHAEL NORIODE** :sunglasses:

File: src/components/App.js (hunk 2 above)
Comment (trox115) [reply src/components/App.js:37]: Hello, maybe I misunderstood this but I should only pass Total or ( not and)  next. And the value variable receives the Total or next as defined on line 28 to 33.

**Review by Oghenebrume50 — APPROVED** [review]
### STATUS: APPROVED

Hi @trox115  good job man. everything works fine, and I like what you did with the props passed to display, the traditional way was to pass next and total separately but this works.
Please merge and move forward **this is approved** 

**RAPHAEL NORIODE** :sunglasses:

**Pull request merged**